clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -429,21 +492,40 @@ public void onResponse(final ClientResponse response) {
                     long offset = offsetAndMetadata.offset();
                     Errors error = Errors.forCode(partition.errorCode());
                     if (error == Errors.NONE) {
-                        log.debug("OffsetCommit {} for partition {}", offset, tp);
+                        log.debug("OffsetCommit completed successfully for offset {} partition {}", offset, tp);
                         continue;
                     }
 
-                    if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
+                    if (error == Errors.COORDINATOR_NOT_AVAILABLE ||
+                        error == Errors.NOT_COORDINATOR ||
+                        error == Errors.REQUEST_TIMED_OUT) {
+                        coordinatorRequestManager.markCoordinatorUnknown(error.message(), currentTimeMs);
+                        retry(currentTimeMs, error.exception());
+                        return;
+                    } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS
+                        || error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
+                        // just retry
+                        retry(currentTimeMs, error.exception());
+                        return;
+                    } else if (error == Errors.UNKNOWN_MEMBER_ID || error == Errors.STALE_MEMBER_EPOCH) {
```

Comment:
As discussed offline, I am not sure about the `UNKNOWN_MEMBER_ID` handling here.

Reply:
Agree, UNKNOWN_MEMBER_ID should end up failing the request. STALE_MEMBER_EPOCH being the one that can be retried if the member got a new one.